eodag/api/search_result.py

```diff
@@ -42,6 +47,43 @@ def crunch(self, cruncher, **search_params):
         crunched_results = cruncher.proceed(self, **search_params)
         return SearchResult(crunched_results)
 
+    def filter_date(self, start=None, end=None):
+        """Apply FilterDate crunch, check its documentation to know more"""
+        return self.crunch(FilterDate(dict(start=start, end=end)))
+
+    def filter_latest_intersect(self, geometry):
+        """Apply FilterLatestIntersect crunch, check its documentation to know more"""
+        return self.crunch(FilterLatestIntersect({}), geometry=geometry)
+
+    def filter_latest_by_name(self, name_pattern):
+        """Apply FilterLatestByName crunch, check its documentation to know more"""
+        return self.crunch(FilterLatestByName(dict(name_pattern=name_pattern)))
+
+    def filter_overlap(
+        self,
+        geometry,
+        minimum_overlap=None,
+        contains=None,
+        intersects=None,
+        within=None,
+    ):
+        """Apply FilterOverlap crunch, check its documentation to know more"""
+        return self.crunch(
+            FilterOverlap(
+                dict(
+                    minimum_overlap=minimum_overlap,
+                    contains=contains,
+                    intersects=intersects,
+                    within=within,
+                )
+            ),
+            geometry=geometry,
+        )
+
+    def filter_property(self, operator=None, **search_property):
```

Comment:
This method `filter_property` delegates to `FilterProperty`. The default values are managed in the class, by using `None` the default values inside `FilterProperty` are used. 
If we add "eq" here there will be redundancy with the implementation of `FilterProperty`. It will help users to know what is the value used by `filter_property` without going to check the doc of `FilterProperty`, but if we change the default value inside `FilterProperty` we will need to also change it here, across multiple files, for one value changed.